scan-payment/src/main/java/com/getbouncer/scan/payment/Image.kt

```diff
@@ -241,6 +241,255 @@ fun Bitmap.zoom(
+            r2 = Pair(y - centerSize.height / 2 - totalHeight / 2, y - centerSize.height / 2)
+            r3 = Pair(y - centerSize.height / 2, y + centerSize.height / 2)
+            r4 = Pair(y + centerSize.height / 2, this.height)
+            c1 = Pair(0, x - centerSize.width / 2 - totalWidth / 2)
+            c2 = Pair(x - centerSize.width / 2 - totalWidth / 2, x - centerSize.width / 2)
+            c3 = Pair(x - centerSize.width / 2, x + centerSize.width / 2)
+            c4 = Pair(x + centerSize.width / 2, this.width)
+
+            if (y == this.height / 2) {
+                r1 = r2
+            }
+
+            if (x == this.width / 2) {
+                c1 = c2
+            }
+
+            tr1 = Pair(
+                toBorderWidth + toCenterDimension + (r1.second - r1.first) * toBorderWidth /
+                    (totalHeight / 2),
+                (2 * toBorderWidth) + toCenterDimension
+            )
+            tr2 = Pair(0, toBorderWidth)
+            tr3 = Pair(toBorderWidth, toBorderWidth + toCenterDimension)
+            tr4 = Pair(
+                toBorderWidth + toCenterDimension,
+                toBorderWidth + toCenterDimension + (r1.second - r1.first) * toBorderWidth /
+                    (totalHeight / 2)
+            )
+            tc1 = Pair(
+                toBorderWidth + toCenterDimension + (c1.second - c1.first) * toBorderWidth /
+                    (totalWidth / 2),
+                (2 * toBorderWidth) + toCenterDimension
+            )
+            tc2 = Pair(0, toBorderWidth)
+            tc3 = Pair(toBorderWidth, toBorderWidth + toCenterDimension)
+            tc4 = Pair(
+                toBorderWidth + toCenterDimension,
+                toBorderWidth + toCenterDimension + (c1.second - c1.first) * toBorderWidth /
+                    (totalWidth / 2)
+            )
+
+            if (y == this.height / 2) {
+                tr1 = tr2
+            }
+        }
+
+        if (x == this.width / 2) {
+            tc1 = tc2
+        }
+    }
+
+    val fromSegments = arrayOf(
+        Rect(c1.first, r1.first, c1.second, r1.second),
+        Rect(c2.first, r1.first, c2.second, r1.second),
+        Rect(c3.first, r1.first, c3.second, r1.second),
+        Rect(c4.first, r1.first, c4.second, r1.second),
+        Rect(c1.first, r2.first, c1.second, r2.second),
+        Rect(c2.first, r2.first, c2.second, r2.second),
+        Rect(c3.first, r2.first, c3.second, r2.second),
+        Rect(c4.first, r2.first, c4.second, r2.second),
+        Rect(c1.first, r3.first, c1.second, r3.second),
+        Rect(c2.first, r3.first, c2.second, r3.second),
+        Rect(c3.first, r3.first, c3.second, r3.second),
+        Rect(c4.first, r3.first, c4.second, r3.second),
+        Rect(c1.first, r4.first, c1.second, r4.second),
+        Rect(c2.first, r4.first, c2.second, r4.second),
+        Rect(c3.first, r4.first, c3.second, r4.second),
+        Rect(c4.first, r4.first, c4.second, r4.second)
```

Comment:
this is 16 regions, but we're only using a 9x9 grid to perform the resize. This means we're doing 70% more image concatenation operations than we would with a 9x9 grid.

Let's instead break this problem down a little more and keep our 9x9.